src/live_restore/live_restore_fs.c

```diff
@@ -529,6 +536,47 @@ __read_promote(WT_LIVE_RESTORE_FILE_HANDLE *lr_fh, WT_SESSION_IMPL *session, wt_
     return (0);
 }
 
+/*
+ * __live_restore_validate_size --
+ *     Validate a read or write is a multiple of the destination file system block size. File
+ *     systems always write data in blocks so when live restore performs a one byte write to a file
+ *     the file system instead writes a block (e.g. 4KB) to disk and which is tracked in the file
+ *     metadata. Live restore thinks we've removed a 1 byte hole while the file metadata says we've
+ *     removed a 4KB hole. When we reopen the file and recover hole information from the metadata it
+ *     will be missing holes and we'll read try data from the destination that was never copied from
+ *     source. Forcing writes to be a multiple of the block size resolves this issue.
+ */
+static int
+__live_restore_validate_size(
```

Comment:
Yep very good point. 
Between this and @sueloverso's comment on the ticket I think the better approach will be increasing the bounds of the read/write to always read full blocks rather than fixing the block size. 

I'll put this PR back to draft while I think through the inputs.